### PR TITLE
Docs rewrite

### PR DESCRIPTION
**Pull request by wacii**
The documentation has continued to grow until it is now difficult to learn from, reference, and maintain.

And so I am going to split up the current documentation into multiple pages and put it up as a website. As is often done with open source projects. During the process I have found some topics I felt weren't properly explained, so I have added some content as well.

There is still a good amount of work to be done, but I thought I'd put up what I've done so far so people can start giving feedback and make requests and whatnot.

### CODE REVIEW

**Comment by wacii** [comment]
Think I'm just going to split off some of the content to start. Then maybe do a website later.

**Comment by sorodrigo** [comment]
Sounds good, at work we started using gitbooks the last week and I've got to say it's pretty neat.

**Comment by wacii** [comment]
@sorodrigo Yeah, I was thinking of using that as that's what Redux uses. Had some trouble getting the links to work. Tried to use the cli for faster feedback but that didn't work on the newest version either. Figure I'll give them some time to get out of alpha/beta before I try again.

I need to add some executable code for the code examples I introduced, and I may want to shorten the quickstart on the main readme, but otherwise it's getting close to what I had imagined.

**Comment by sorodrigo** [comment]
I found out today that they had gone into beta. And it's still super green, I felt a little embarrassed for recommending it (I meant to recommend the legacy version 😂). But loving the news, I hope/expect I can spend some time on this on Thursday, so I'll tackle the PRs then.

**Comment by wacii** [comment]
Alright, I think this is ready for review.

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+  * [Write Resilience](basics/write-resilience.md)
+  * [Rollback](basics/rollback.md)
+  * [Commit](basics/commit.md)
+* Recipes
+  * [Customize Requests](recipes/customize-requests.md)
+  * [Immutable.js](recipes/immutable.md)
+  * [Redux-Saga](recipes/redux-saga.md)
+* API Reference
+  * [offline()](api/offline.md)
+  * [createOffline()](api/create-offline.md)
```
Comment (sorodrigo) [inline docs/README.md:16]: Broken link should be either `createOffline.md` or renaming that file.

**Comment by sorodrigo** [comment]
Kudos on the docs rewrite!! 🎉 

I don't have an easy way of editing your PRs because you work directly on your fork. It would be easier for me if you could push your branches to this repo, I think you already have the push/merge access.

**Pull request merged**